R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+#'                   use analysisSofteware to connect data_input and data_analysis
+#' @param data_analysis Software tools used to generate the QC metrics
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+getmetaData <- function(input, data_input, data_analysis){
+  inputFile <- list()
+  analysis <- list()
+  input <- unique(input)
+  for (x in (1:length(input))) {
+    locate <- which(input[x] == data_input$name)
+    inputFile[x] <- list(data_input[locate, ][1:3])
+    analysis[x] <- list(data_analysis[which(data_input[locate,]$analysisSoftware == data_analysis$name), ])
+  }
+  analysis <- unique(analysis)
+  
+  metaData = list(inputFiles = list(inputFile = inputFile), analysisSoftware = analysis)
+  return(metaData)
+}
+
+#'
+#' Get the Qualities(either runQualities or setQualities) composed of metadata and qualityMetrics
+#' 
+#' 
+#' 
+#' @param Qualities the filtered data frame of metric
+#' @param data_input File based on which QC metrics were generated
+#'                   use analysisSofteware to connect data_input and data_analysis
+#' @param data_analysis Software tools used to generate the QC metrics
+#' @param set judge whether the quality is setQuality
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+getQualities <- function(Qualities, data_input, data_analysis, set){
+  if(length(Qualities[ , 1]) != 0 ){
+    if(set == TRUE){
+      input <- as.list(Qualities$input_file)
+      input_file <- unlist(lapply(input, function(x){
+        if (grepl(",", x)){
+          input[which(x == as.list(Qualities$input_file))] <- NULL
+          input <- unlist(c(input, unlist(strsplit(x , split = ","))))
+        }
+      }))
+    }else{
+      input_file <- as.list(Qualities$input_file)
+    }
+    metaData <- getmetaData(input = input_file, data_input, data_analysis)
+    Qualities <- Qualities[! is.na(Qualities$value), ]
+    Qualities <- list(metaData = metaData, qualityMetrics = list(qualityMetric = transform_data(data = Qualities[1:5])))
+    return(Qualities)
+  }
+}
+
+
+#'
+#' Write all mzQC related information to the JSON file
+#' 
+#' 
+#' 
+#' @param version version of mzQC
+#' @param creationDate JSON file generation time
+#' @param runQualities data frame of runQuality elements
+#' @param setQualities data frame of setQuality elements
+#' @param controlledVocabularies use to refer to the source of the used cv terms in the qualityMetric objects
+#' @param file output file path
+#' @param data_input File based on which QC metrics were generated
+#'                   use analysisSofteware to connect data_input and data_analysis
+#' @param data_analysis Software tools used to generate the QC metrics
+#' @return a list
+#' 
+#' @export
+#' 
+#' 
+#' 
+createmzQC <- function(version, creationDate, runQualities, setQualities, controlledVocabularies, file, data_input, data_analysis){
+  ##if data frame of runQualities is empty, set runQualities to list()
+  if(length(runQualities[ ,1]) == 0){
```

Comment:
if `getQualities()` would check for `length(runQualities[ ,1]) == 0)` internally, then you can get rid of all this `if .. else ... else` code